code/retrieval/qdrant.py

```diff
@@ -705,4 +705,90 @@ async def search_all_sites(self, query: str, num_results: int = 50,
             List[List[str]]: List of search results
         """
         # This is just a convenience wrapper around the regular search method with site="all"
-        return await self.search(query, "all", num_results, collection_name, query_params)
\ No newline at end of file
+        return await self.search(query, "all", num_results, collection_name, query_params)
+    
+    async def get_sites(self, collection_name: Optional[str] = None) -> List[str]:
+        """
+        Get a list of unique site names from the Qdrant collection.
+        
+        Args:
+            collection_name: Optional collection name (defaults to configured name)
+            
+        Returns:
+            List[str]: Sorted list of unique site names
+        """
+        collection_name = collection_name or self.default_collection_name
+        logger.info(f"Retrieving unique sites from collection: {collection_name}")
+        
+        try:
+            client = await self._get_qdrant_client()
+            
+            # Check if collection exists
+            if not await client.collection_exists(collection_name):
+                logger.warning(f"Collection '{collection_name}' does not exist")
+                return []
+            
+            # Use scroll to get all points with site field
+            sites = set()
+            offset = None
+            batch_size = 1000
+            
+            while True:
+                try:
+                    points, next_offset = await client.scroll(
+                        collection_name=collection_name,
+                        limit=batch_size,
+                        offset=offset,
+                        with_payload=["site"]
+                    )
+                    
+                    if not points:
+                        break
+                    
+                    # Extract site values
+                    for point in points:
+                        site = point.payload.get("site")
+                        if site:
+                            sites.add(site)
+                    
+                    offset = next_offset
+                    if offset is None:
+                        break
+                        
+                except Exception as e:
+                    if "Collection not found" in str(e):
```

Comment:
I think this would be an unreachable condition. Due to the check at L#727.
I propose removing this try, except block.